src/wallet/api.rs

```diff
@@ -884,6 +885,28 @@ impl Wallet {
         Ok((privkey, public_key))
     }
 
+    /// Refreshes the UTXO cache using the wallet's internal spend info checks.
+    pub(crate) fn refresh_utxo_cache(&mut self, utxos: Vec<ListUnspentResultEntry>) {
```

Comment:
Unify all the functions into one wallet function `update_utxo_cache` and do all the processing there. Having the logic spread into multiple functions makes it harder to reason about it later on.